src/node/mod.rs

```diff
@@ -160,11 +157,148 @@ impl Node {
+
+        // Split class name by dots and push as path segments for loading the file.
+        if let Some((path, cn)) = cls.rsplit_once('.') {
+            for p in path.split('.') {
+                cpath.push(p);
+            }
+            cpath.push(format!("{cn}.yml"));
+        } else {
+            cpath.push(format!("{cls}.yml"));
+        }
+
+        // Extract the directory in which the new class is stored to use for the new class's
+        // `own_loc`
+        let class_loc = if let Some(parent) = cpath.parent() {
+            parent
+                .strip_prefix(&r.classes_path)
+                .map_err(|e| anyhow!("Error while rendering class location for {}: {}", class, e))?
+        } else {
+            return Err(anyhow!(
+                "Can't determine class location in hierarchy for {}",
+                class
+            ));
+        };
+
+        let mut meta = NodeInfoMeta::default();
+        let (ccontents, cpath) = load_file(&cpath)?;
+        meta.uri = format!("yaml_fs://{}", std::fs::canonicalize(cpath)?.display());
+        let c = Node::from_str(meta, Some(PathBuf::from(class_loc)), &ccontents)
+            .with_context(|| format!("Deserializing {cls}"))?;
+        Ok(c)
+    }
+
+    /// Merges self into other, then updates self with merged values from other
+    fn merge_into(&mut self, other: &mut Self) -> Result<()> {
+        // We use std::mem::take() here so we can merge self.applications into other.applications
+        // without having to call `clone()` twice. This doesn't destroy `self.applications` because
+        // we update `self.applications` with the result of the merge immediately afterwards.
+        let self_apps = std::mem::take(&mut self.applications);
+        other.applications.merge(self_apps);
+        self.applications = other.applications.clone();
+
+        // We use std::mem::take() here so we can merge self.classes into other.classes without
+        // having to call `clone()` twice. This doesn't destroy `self.classes` because we update
+        // `self.classes` with the result of the merge immediately afterwards.
+        let self_classes = std::mem::take(&mut self.classes);
+        other.classes.merge(self_classes);
+        self.classes = other.classes.clone();
+
+        other.parameters.merge(&self.parameters)?;
+        self.parameters = other.parameters.clone();
+        Ok(())
+    }
+
+    /// Recursively loads classes and merges loaded data into self
+    fn render_impl(&mut self, r: &Reclass, seen: &mut Vec<String>, root: &mut Node) -> Result<()> {
+        for cls in self.classes.items_iter() {
+            if seen.contains(cls) {
+                continue;
+            }
+
+            // TODO(sg): parse and render references in class names
+            let cls = cls.to_string();
+
+            let maybec = self.read_class(r, &cls);
+            let mut c = if let Ok(c) = maybec {
+                c
+            } else if r.ignore_class_notfound {
```

Comment:
This would also ignore other errors that could happen now/ in the future?

Reply:
Probably, I didn't fully think this through. It's probable better to move the `ignore_class_notfound` to `read_class()` and have that method return `Result<Option<Node>>`

Reply:
Side-note: I found a bug (relative to Python reclass) while refactoring the class notfound flag handling. We now discover classes the same way that Python reclass does -- by walking the classes_path and storing a map between class names and file paths.